crypto/evp/e_aes.c

```diff
@@ -2717,7 +2717,9 @@ static int aes_gcm_init_key(EVP_CIPHER_CTX *ctx, const unsigned char *key,
         do {
 #ifdef HWAES_CAPABLE
             if (HWAES_CAPABLE) {
-                HWAES_set_encrypt_key(key, ctx->key_len * 8, &gctx->ks.ks);
+                HWAES_set_encrypt_key(key,
+                                      EVP_CIPHER_CTX_get_key_length(ctx) * 8,
```

Comment:
This function can return an error.. Same comment applies to many places you have added this accessor

Reply:
Oops :(

Reply:
I've added failure checks and cleaned up some of the pointer comparisons.